src/test/java/org/example/EchoServerTest.java

```diff
@@ -23,16 +23,17 @@ public void testServerSocketGetsCreated() throws IOException {
     @Test
     public void testServerSocketCreatedWithCorrectPort() throws IOException {
         int portNumber = 7777;
-        ServerSocket testServerSocket = echoServerTest.serverSocketCreator(portNumber);
+        try (ServerSocket testServerSocket = echoServerTest.serverSocketCreator(portNumber)) {
 
-        assertEquals(testServerSocket.getLocalPort(), portNumber);
+            assertEquals(testServerSocket.getLocalPort(), portNumber);
+        }
     }
     @Test
     public void testClientSocketGetsCreated() throws IOException {
         ServerSocket mockServerSocket = mock(ServerSocket.class);
         when(mockServerSocket.accept()).thenReturn(new Socket());
         assertNotNull(
-                echoSocketTest.socketCreator(mockServerSocket));
+                clientSocketTest.socketCreator(mockServerSocket));
     }
```

Comment:
It might be overkill, but there is a way in [Mockito where you can verify](https://www.baeldung.com/mockito-verify) how many times a method has been called.

Reply:
Added a `verify` assertion to the latest commit.